app/models/concerns/product_sort_by_stocks.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module ProductSortByStocks
+  extend ActiveSupport::Concern
+
+  included do
+    @on_hand_sql = Arel.sql("(
+      SELECT COALESCE(SUM(si.count_on_hand), 0)
+      FROM spree_variants v
+      JOIN spree_stock_items si ON si.variant_id = v.id
+      WHERE v.product_id = spree_products.id
+      GROUP BY v.product_id
+    )")
+
+    @backorderable_priority_sql = Arel.sql("(
+      SELECT CASE WHEN BOOL_OR(si.backorderable) = true THEN 1 ELSE 0 END
```

Comment:
You might have tried this already, but I wonder if this works with the simple boolean value:
```suggestion
      SELECT BOOL_OR(si.backorderable)
```

PS I didn't know about BOOL_OR, that's a handy function!